framework/src/wifi_manager/wifi_manager.c

```diff
@@ -395,8 +422,19 @@ wifi_manager_result_e wifi_manager_init(wifi_manager_cb_s *wmcb)
 	g_manager_info.ip4_address[0] = '\0';
 	g_manager_info.rssi = 0;
 	g_manager_info.mode = STA_MODE;
+	g_manager_callback = wmcb;
 	wifi_status_set(AP_DISCONNECTED);
-	g_manager_info.wmcb = wmcb;
+
+	if (wifi_utils_get_info(&info) != WIFI_UTILS_SUCCESS) {
```

Comment:
To reduce duplicated code and handle mutex, How about create one goto?
Below can be one of example.
result = wifi_utils_get_info(&info);
if (result != WIFI_UTILS_SUCCESS) {
     dgb();
     goto error_with_mutex;
}
.
.
error_with_mutex:
    wifi_mutex_release(w_mutex);
    free(w_mutex);
    w_mutex = NULL;
    if (result != WIFI_UTILS_SUCCESS) {
         return WIFI_MANAGER_FAIL;
    }
    return WIFI_MANAGER_SUCCESS;

Reply:
I accept your recommendation.